client/src/lib/fetch.js

```diff
@@ -0,0 +1,18 @@
+import { SERVER_URL } from "../constants";
+
+export const getUserData = async (id = "") => {
```

Comment:
What does the default value `""` mean? If we should never call getUserData without an ID, then consider removing the default value. Alternatively, consider using `"me"` as the default.

Also, stylistically I have a small preference for using regular async functions rather than arrow (fewer characters!). Up to you though.

```suggestion
export async function getUserData(id) {
```